### PR TITLE
OCPBUGS-3987: Check nmstateconfig content in agent-config.yaml

### PR DESCRIPTION
**Pull request by bfournie**
The NMStateConfig data in agent-config.yaml was not being validated, it was only checked in the Load() i.e. the data in nmstateconfig.yaml. This can result in invalid configurations being stored in ignition and then the install would fail when these configs were validated by assisted-service.

The change is to do the same checks in Load() and Generate().

### CODE REVIEW

**Review by pawanpinjarkar — COMMENTED** [review]
/lgtm

File: pkg/asset/agent/manifests/nmstateconfig.go [hunk 1]
```diff
@@ -199,8 +189,18 @@ func (n *NMStateConfig) Load(f asset.FileFetcher) (bool, error) {
 
 func (n *NMStateConfig) finish() error {
 
-	if err := n.validateNMStateConfig().ToAggregate(); err != nil {
-		return errors.Wrapf(err, "invalid NMStateConfig configuration")
+  level := logrus.GetLevel()
+	logrus.SetLevel(logrus.WarnLevel)
+	staticNetworkConfigGenerator := staticnetworkconfig.New(logrus.WithField("pkg", "manifests"), staticnetworkconfig.Config{MaxConcurrentGenerations: 2})
+	defer logrus.SetLevel(level)
```
[the remark this answers is not in the document]
Comment (bfournie) [reply pkg/asset/agent/manifests/nmstateconfig.go:207]: Yep, moved to a separate function.

**Pull request merged**